### ISSUE TITLE
Split out timeseries dataframe base class

### ISSUE THREAD

**Issue by znicholls** [openclimatedata/openscm#194]
In discussions with @lewisjared, we thought it would be helpful to split out `ScmDataFrameBase` (maybe with a new name) into a new package. Then it could be imported in other places (e.g. Pymagicc) without having OpenSCM as an explicit dependency. That would also fix our current circular dependency where OpenSCM depends on Pymagicc and vice versa.

Thoughts @swillner and @rgieseke ?

**Comment by rgieseke** [comment]
Could be useful! What is the relation to Pyam's IamDataFrame? Would it depend on that?

**Comment by znicholls** [comment]
Nope. You could cast to IamDataFrame to get access to their plotting
capabilities but it wouldn’t be a direct dependency (same as now).
